### DISCUSSION TITLE
flash.system.System.output()

### DISCUSSION THREAD

**Discussion by MalacTheLittle** [airsdk/Adobe-Runtime-Support#1274] [Q&A]
Hi!
Can someone point me how to use new `output` method?
As I understand, if I run .exe application from command prompt, output method should echo entered `String` to command prompt. When I use it like that, nothing is echoed...

What am I doing wrong?

**Comment by marchbold** [comment]
I just use it as below without issues in APM:

```actionscript
System.output( "some string" );
```

https://github.com/airsdk/apm/blob/master/client/src/com/apm/client/io/IO_MacOS.as#L114

What exactly have you tried?

**Comment by ajwfrost** [comment]
One thought .. you mention that you're running it as an .exe, does this mean you've used the ADT packaging command to create an application? If so, it will be a Windows GUI application, and would not automatically have any console output. I can ask someone to double-check this, but it may be we would need to update the code to check whether the app was launched from a console, and if so we could hopefully connect to it. We just need to make sure that GUI-based apps (where you just click on an icon to launch them) are not then suddenly triggering console windows to open spuriously!

What are you using the output for though? as there may be better ways of achieving the same sort of goal...
 
Final question that's just been pointed out here: what version of the AIR runtime are you using? If the application .exe file has been packed with the captive runtime then there's a folder alongside it called "Adobe AIR" and within this, "Versions\1.0\Adobe AIR.dll" - what's the version details on that DLL? or if you're using a shared runtime, what's the version in the "C:\Program Files (x86)\Common Files\Adobe AIR\Version\1.0\Adobe AIR.dll" file? I think all this is in AIR 33.1.1..533 and later..

**Reply by MalacTheLittle** [reply]
You're right, I'm using it on Windows and I've used ADT packaging command, so it's bundle app. 

I'm trying to make app so user can run it either by double click on icon and from command prompt. If user runs it (a) by double click, it runs normally with GUI, and if user runs it (b) from command prompt with arguments e.g. `-clean myFile.json`, app doesn't show it's GUI, but just make some changes in requested file, outputs some message to command prompt like "operation successfull" or "error: file not found" , and turns off. This part I would manage with function associated by InvokeEvent.

I tried it with AIR 33.1.1.554 and AIR 33.1.1.633 with no success.

Now I've tried it on MacOS like @marchbold is using it, and it works like a charm :) 
It would be great to work on Windows too.

**Reply by ajwfrost** [reply]
Ah - so yes, thanks, probably the console connection. Let me check with my colleague..

**Reply by MalacTheLittle** [reply]
Forgot to answer, .dll's are same version as AIR SKD's... I gave it another shot with AIR 33.1.1.533, same result :/

Since you can read `standardError` and `standardOutput` from `NativeProcess`, it would be great to be able to write these two back to `System` so other apps that weren't made in AIR could have same benefit by getting these "messages".

**Reply by marchbold** [reply]
Ah yeah we haven't tried packaging as an exe yet on windows. Still running with `adl` directly. That was on the list for next month :)

**Reply by MalacTheLittle** [reply]
Perfect! Looking forward to it :)
I must say you're doing pretty awesome job here making us AIR survivors proud for still standing! There's a lot of great AIR apps behind us, and obviously a lot in front of us :)

**Reply by marchbold** [reply]
Cheers! Hopefully we can pull it back from the last years of stagnation hey! :)
